### ISSUE TITLE
Searching post by URL finds nothing when logged out

### ISSUE THREAD

**Issue by nemobis** [mastodon/mastodon#28376]
### Steps to reproduce the problem

1. Visit a Mastodon v4.2.3 instance on the web, for example mastodon.social or mastodon.world
2. Search a public post like https://framapiaf.org/@wikimediaitalia/111535119898885068 or https://mastodon.social/@Mastodon/111461309876030410
3. Get "Could not find anything for these search terms"
4. Repeat the search while logged in
5. Get the post
...

### Expected behaviour

Search by URL should retrieve the post (and cache it if it's not already)

### Actual behaviour

Nothing found

### Detailed description

Is this intentional behaviour? I expect it to happen with AUTHORIZED_FETCH enabled, but that's not the case on framapiaf.org or mastodon.social AFAIK.

I was trying to retrieve all the posts which were part of a thread, specifically https://mastodon.social/@rmounce/109789740766182785 which is the parent of https://mastodon.social/@pintoch@mamot.fr/109789773221771207  but doesn't show up on https://mamot.fr/@pintoch/109789772868585765 (for presumably unrelated reasons).

### Mastodon instance

mastodon.social

### Mastodon version

v4.2.3

### Browser name and version

Firefox 120

### Operating system

Fedora Linux

**Comment by nemobis** [comment]
This sounds like a duplicate of https://github.com/mastodon/mastodon/issues/23196

**Comment by nemobis** [comment]
If this is an intentional feature, it's actually a first step towards fixing https://github.com/mastodon/mastodon/issues/22620 and https://github.com/mastodon/mastodon/issues/22924.

**Comment by ciencia** [comment]
Displaying local posts should be fine IMHO (#23196), but retrieving external posts through search for logged-out users may not be desirable, since this probably caches locally the post and attached media, which could be used to abuse the instance by searching for objectionable/malicious content that may later appear in the federated feed, or filling-up the local instance disk with media and posts which normally wouldn't be retrieved by requesting a large amount of external posts.

**Comment by nemobis** [comment]
Indeed. This can be a feature, in which case the bug is that the error message says "no results found" instead of saying that you need to log in to look up the post.
